config/rbac/operator-role-knative.yaml

```diff
@@ -49,6 +51,8 @@ rules:
   - messaging.knative.dev
   resources:
   - subscriptions
+  - channels
+  - inmemorychannels
```

Comment:
this is being fixed by the proper addressable-resolver cluster role binding in #3400

Reply:
So it is no longer necessary to add these to the permissions and can be removed from this PR?

Reply:
yes, this is obsolete and you can remove that because it is already covered by the addressable-resolver cluster role provided by Knative itself. In the end Camel K just uses a cluster role binding in order to bind the operator service account to this addressable-resolver cluster role. This should give all read permissions to Knative resources such as brokers, channels, etc.

Not sure about these create/delete permissions that you have also added in this PR for brokers and routes. Do we need that for some of the e2e tests?

Reply:
@christophd 
All additions to the permissions have been required due to tests failure on OCP4 and log messages to the affect that 'systemaccount... was forbidden from ${verb}ing x:y:z'

Reply:
@phantomjinx yep makes sense, I wonder though if the write permissions are only required for the e2e tests. I can not recall that Camel K creates Knative brokers in production code